enioka_scan/src/main/java/com/enioka/scanner/api/Scanner.java

```diff
@@ -70,107 +71,213 @@ default void releaseScanTrigger() {
 
     /**
      * The app keeps the scanner for itself but does not need it immediately. It may free whatever resources it has, or ignore this call.
+     * FIXME: may be considered a feature not supported by all (e.g. GsSppScanner does not support it, PostechSppScanner may not support it)
      */
     void pause();
 
     /**
      * Reverse the effects of {@link #pause()}. The scanner is once again ready to scan after this call. Status callback should be called if needed. Idempotent.
+     * FIXME: may be considered a feature not supported by all (e.g. GsSppScanner does not support it, PostechSppScanner may not support it)
      */
     void resume();
 
 
     ////////////////////////////////////////////////////////////////////////////////////////////////
-    // BEEPS
+    // FEATURE SUPPORT
     ////////////////////////////////////////////////////////////////////////////////////////////////
 
     /**
-     * Short high beep to indicate successful scan
+     * Casts the current scanner to the WithBeepSupport interface if the feature is supported.
+     * @return `this` if the feature is supported, or `null` if the feature is not supported.
      */
-    void beepScanSuccessful();
+    default @Nullable WithBeepSupport getBeepSupport() {
+        if (this instanceof WithBeepSupport) {
+            return (WithBeepSupport) this;
+        }
+        return null;
+    }
 
     /**
-     * Long low beep to indicate unsuccessful scan
+     * Casts the current scanner to the WithTriggerSupport interface if the feature is supported.
+     * @return `this` if the feature is supported, or `null` if the feature is not supported.
      */
-    void beepScanFailure();
+    default @Nullable WithTriggerSupport getTriggerSupport() {
```

Comment:
optional

Reply:
Those appear to require API 24 at least, we use API 19 as our min requirement so they are not supported.